frontend/src/pages/Register.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from "react";
+import { TextField, Button, Container, Typography } from "@mui/material";
+import registerClients from "../hooks/registerClients";
+
+const RegisterForm = () => {
+  const { addClient } = registerClients();
+  const [email, setEmail] = useState<string>("");
+  const [payment_address, setPaymentAddress] = useState<string>("");
+  const [delivery_address, setDeliveryAddress] = useState<string>("");
+  const [nip, setNip] = useState<string>("");
+  const [orders, setOrders] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [company_name, setCompanyName] = useState<string>("");
+  const [error, setError] = useState<string>("");
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    // Walidacja
+    if (!email || !password || !company_name || !nip || !payment_address || !delivery_address) {
+      setError("Proszę wypełnić wszystkie pola");
+      return;
+    }
+
+    setError("");
+
+    const newClient = {
+      email,
+      payment_address,
+      delivery_address,
+      nip,
+      orders,
+      password,
+      company_name,
+    };
+
+    console.log("Form data to be sent:", newClient);
+
+    try {
+      await addClient(newClient);
+      console.log("Zarejestrowano:", newClient);
```

Comment:
for development purposes we can leave this log while finish this PR but before merging it we need to delete it. Also as said before we cannot register user on backend and then resend his password and other privacy data here, he should get a token instead